skimage/meson.build

```diff
@@ -29,37 +29,61 @@ if is_windows
   endif
 endif
 
-# NumPy include directory - needed in all submodules
-incdir_numpy = run_command(py3,
-  [
-    '-c',
-    'import os; os.chdir(".."); import numpy; print(numpy.get_include())'
-  ],
-  check: true
-).stdout().strip()
+# When cross-compiling skimage, the compiler needs access to NumPy and Pythran
+# headers for the host platform (where the package will actually run). These
+# headers may be incompatible with any corresponding headers that might be
+# installed on the build system (where the compilation is performed). To make
+# sure that the compiler finds the right headers, paths can be configured in
+# the 'properties' section of a Meson cross file:
+#
+#   [properties]
+#   numpy-include-dir = '/path/to/host/numpy/includes'
+#   pythran-include-dir = '/path/to/host/pythran/includes'
```

Comment:
Probably a stupid question, but are these paths intentionally different from `incdir_numpy` and `incdir_pythran` which are actually used below?

Reply:
If you mean the property names themselves, I used the same names as in SciPy for consistency and because they are more descriptive than `incdir_*` for somebody writing a cross file. If you mean the values, they will be different during cross compilation. Executing Python and querying the modules will generally return a system path like

    /usr/lib/python3.11/site-packages/numpy/core/include

while a cross environment might require a path like

    /usr/${arch_triple}/usr/lib/python3.11/site-packages/numpy/core/include

(This is the scheme we use when cross-compiling packages for Void Linux, where `${arch_triple}` might be something like `arm-linux-gnueabif`.)

Reply:
I was just confused about the property names themselves and couldn't find the origin of these names. Am I right to assume that the only requirement for these names is that they have to match whatever a potential cross build definition file would expect?

Reply:
I believe they are arbitrary; Meson just stores the properties until `meson.get_external_property` attempts to read them. Thus, `meson.build` determines the form they should take, and any cross file needs to conform to the expectation.

Reply:
Okay, I think I get it. Thanks for explaining patiently! :slightly_smiling_face: